### ISSUE TITLE
Schedule Free Optimizer (SGD, AdamW)

### ISSUE THREAD

**Issue by vasqu** [huggingface/transformers#30087]
### Feature request

See [schedule_free](https://github.com/facebookresearch/schedule_free): Schedule free optimizer alternatives for adamW and SGD. If possible, without needing it as a separate dependency.

### Motivation

Getting the right optimizers and even schedulers can be a very difficult and annoying task. Allowing scheduler-free approaches might help in reducing finding the right schedules. Quoting the meta repo:
"TLDR Faster training without schedules - no need to specify the stopping time/steps in advance!"

### Your contribution

Not sure if could help but I'd suggest looking at the [Caveats](https://github.com/facebookresearch/schedule_free?tab=readme-ov-file#caveats) especially (e.g. BatchNorm issue, default values, [minimum learning rate](https://github.com/facebookresearch/schedule_free/issues/5)..).

**Comment by vasqu** [comment]
Closing this issue since #30079 will take care of it
